Assets/Scripts/Gameplay/EnergyLabel.cs

```diff
@@ -0,0 +1,24 @@
+using UnityEngine;
+using TMPro;
+
+public class EnergyLabel : MonoBehaviour
+{
+    public float energy;
+    private TextMeshProUGUI labelText;
+
+    private void Awake()
+    {
+        energy = 100f;
```

Comment:
This can cause weird bugs. If we expose energy as a configurable field from inside the editor, this will run after and overwrite what we set in the editor. I think you can set the default value when you add it as a member `public float energy = 100f;`